platform/migration/platform-migration/src/test/java/org/codice/ddf/configuration/admin/ConfigurationFilesPollerTest.java

```diff
@@ -86,362 +76,198 @@ public void setup() {
-                mockExecutorService);
+                configFilePath,
+                FILE_EXT) {
+            @Override
+            void waitForFileToBeCompletelyWritten(File file) throws InterruptedException {
+                return;
+            }
+        };
+        when(configFilePath.toAbsolutePath()).thenReturn(Paths.get("foo"));
+        File file = Files.createTempFile("foo", "bar")
+                .toFile();
 
         // Perform Test
-        configurationFilesPoller.register(null);
-    }
-
-    /**
-     * Verify that file modification events are not sent to the ChangeListener.
-     */
-    @Test
-    public void testRunFileModifiedChangeListenerNotNotified() throws Exception {
-        // Setup
-        Path mockPath = getMockPath(PID, FILE_EXT);
-        WatchEvent<?> mockWatchEvent = getMockWatchEvent(mockPath,
-                StandardWatchEventKinds.ENTRY_MODIFY);
-        List<WatchEvent<?>> watchEvents = getSingleMockWatchEvent(mockWatchEvent);
-        WatchKey mockWatchKey = getMockWatchKey(watchEvents);
-        WatchService mockWatchService = getMockWatchService(mockWatchKey);
-        Path mockConfigurationDirectory = getMockConfigurationDirectoryPath(mockPath,
-                mock(Path.class));
-        ConfigurationFilesPoller configurationFilesPoller = createConfigurationFilesPoller(
-                mockWatchService,
-                mockConfigurationDirectory);
         configurationFilesPoller.register(mockChangeListener);
-
-        // Perform Test
-        configurationFilesPoller.run();
+        configurationFilesPoller.onFileChange(file);
 
         // Verify
-        verify(mockChangeListener, never()).notify(any(Path.class));
+        verify(mockChangeListener, never()).notify(any());
     }
 
     /**
-     * Verify that overflow events are not sent to the ChangeListener.
+     * Verify that the ChangeListener is not notified on file delete events.
      */
     @Test
-    public void testRunOverflowChangeListenerNotNotified() throws Exception {
+    public void testRunFileDeletedChangeListenerNotNotified() throws Exception {
         // Setup
-        Path mockPath = getMockPath(PID, FILE_EXT);
-        WatchEvent<?> mockWatchEvent = getMockWatchEvent(mockPath,
-                StandardWatchEventKinds.OVERFLOW);
-        List<WatchEvent<?>> watchEvents = getSingleMockWatchEvent(mockWatchEvent);
-        WatchKey mockWatchKey = getMockWatchKey(watchEvents);
-        WatchService mockWatchService = getMockWatchService(mockWatchKey);
-        Path mockConfigurationDirectory = getMockConfigurationDirectoryPath(mockPath,
-                mock(Path.class));
-        ConfigurationFilesPoller configurationFilesPoller = createConfigurationFilesPoller(
-                mockWatchService,
-                mockConfigurationDirectory);
-        configurationFilesPoller.register(mockChangeListener);
+        ConfigurationFilesPoller configurationFilesPoller = new ConfigurationFilesPoller(
```

Comment:
✏️  Instead of inlining this class everywhere, could we create a static inner class and use that instead?